stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -235,7 +290,7 @@ struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
     }
 
     for (size_t i = 0; i < num_vars_shared_terms; ++i) {
-      partials[num_vars_sliced_terms + i] = worker.args_adjoints_(i);
+      varis[num_vars_sliced_terms + i]->adj_ = shared_terms_adjoints(i);
```

Comment:
Same question here about adjoints in the forward pass -- I don't think we can do it.

Reply:
Why not? We save and restore. Nothing changes once the reduce function is done during the forward pass.

... and using the adjoints of the outer AD tape is the main idea of this PR which I thought you agreed with already, no?

Reply:
Yah yah this goes with the other comments I got my wires crossed.